nixos/modules/virtualisation/qemu-vm.nix

```diff
@@ -121,26 +121,29 @@ let
       fi
 
       # Create a directory for exchanging data with the VM.
-      mkdir -p $TMPDIR/xchg
+      mkdir -p "$TMPDIR/xchg"
```

Comment:
i see many changes of the kind that would also be suggested by the tool `shellcheck` - did you run that tool on the result and got the inspiration for the tool from there?

Maybe we could have shellcheck run over the output automatically?
(In that case it might also help if the script would be less concatenated from nix snippets)

Reply:
Yes, I run shellcheck on the `result/bin/run-machine-vm` script and applied the changes to the Nix code.
I guess It could be added as a `checkPhase` to the writer functions (`writeShellScript`, etc.), but I'm not sure what the performance penality would be.

Reply:
shellcheck is usually pretty fast but the problem I with adding that is that we probably need to disable some checks and also add a lot more ignore comments at false positives.